### PR TITLE
Volumes from buffer EPP

### PR DESCRIPTION
**Pull request by Mropat**
### This PR adds | fixes:
- Integreates twist_get_volumes_from_buffer EPP with new modifications

### Review:
- [x] Code approved by @mayabrandi 
- [x] Tests executed by @cesve @mayabrandi 
- [x] "Merge and deploy" approved by @mayabrandi 

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes
- [x] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: cg_lims/EPPs/udf/calculate/twist_get_volumes_from_buffer.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+import click
+
+from cg_lims import options
+from cg_lims.get.artifacts import get_artifacts, get_latest_artifact
+
+LOG = logging.getLogger(__name__)
+
+
+@click.command("get-buffer")
```
Comment (mayabrandi) [inline cg_lims/EPPs/udf/calculate/twist_get_volumes_from_buffer.py:13]:
```suggestion
@click.command("twist-get-buffer")
```

We should decide on some kind of rule for naming EPP commands. I think we want workflow specific commands to indicate what workflow they belong to. We can try like this and maybe we will have to rethink about it in the future when we have more EPPs.

File: cg_lims/EPPs/udf/calculate/twist_get_volumes_from_buffer.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+@options.process_type(help="Get buffer from this process type(s)")
+@click.pass_context
+def get_buffer(ctx, process_type: List[str]) -> None:
+    """Getting Volume Elution from previous step of type defined by process_types.
```
Comment (mayabrandi) [inline cg_lims/EPPs/udf/calculate/twist_get_volumes_from_buffer.py:17]: I have started to involve @cesve when creating the dockstriungs of the EPPs, to get them more lab-informative. Why do we have -10 when creating the volume buffer here eg. I think cissi can help to formulate this for uss.

**Review by mayabrandi — COMMENTED** [review]
Looks great @Mropat ! So happy to have you writing your first piece of lims code! Ill approve when we have a test!

File: cg_lims/EPPs/udf/calculate/twist_get_volumes_from_buffer.py (hunk 2 above)
Comment (cesve) [inline cg_lims/EPPs/udf/calculate/twist_get_volumes_from_buffer.py:17]: """As part of the sample is used in the QC, the value is subtracted by 10 to get the actual volume that is left."""

File: cg_lims/EPPs/udf/calculate/twist_get_volumes_from_buffer.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    If volume found, setting the value Volume udf on artifact of current step."""
```
Comment (cesve) [inline cg_lims/EPPs/udf/calculate/twist_get_volumes_from_buffer.py:18]:
```suggestion
    If volume found, setting the value Volume udf on artifact of current step. As part of the sample is used in the QC, the value is subtracted by 10 to get the actual volume that is left."""
```

**Comment by mayabrandi** [comment]
Configure on stage:

Skärmavbild 2020-11-06 kl  14 53 17


Two samples in the step where we werte testing, did not go throuh buffer exchange.
We dont want the script to want about that but only set the volumes that were found:

Skärmavbild 2020-11-06 kl  14 49 49


The script behaves as expected.


Tested by
@cesve @mayabrandi

**Pull request merged**